packages/@react-aria/grid/src/useGridSelectionAnnouncement.ts

```diff
@@ -96,7 +96,17 @@ export function useGridSelectionAnnouncement<T>(props: GridSelectionAnnouncement
     }
 
     lastSelection.current = selection;
-  }, [selection]);
+  });
+
+  useUpdateEffect(() => {
+    if (state.selectionManager.isFocused) {
+      announceSelectionChange();
+    } else {
+      // Wait a frame in case the collection is about to become focused (e.g. on mouse down).
+      let raf = requestAnimationFrame(announceSelectionChange);
+      return () => cancelAnimationFrame(raf);
+    }
+  }, [selection, state.selectionManager.isFocused]);
```

Comment:
Not necessarily related to this change (but spurred due to the comment of mouse down here) but now that the Table's cell is focused on click rather than the row, the announcement of row selection seems to be interrupted a lot more, at least on Safari 17 + VoiceOver.